autoload/airline/extensions/branch.vim

```diff
@@ -68,37 +71,35 @@ function! s:get_git_branch(path)
 endfunction
 
 function! s:get_git_untracked(file)
-  let untracked = ''
-  if empty(a:file)
-    return untracked
+  if empty(a:file) || !executable('git')
+    return
   endif
-  if has_key(s:untracked_git, a:file)
-    let untracked = s:untracked_git[a:file]
-  else
-    let output    = system('git status --porcelain -- '. shellescape(a:file))
-    if output[0:1] is# '??' && output[3:-2] is? a:file
-      let untracked = get(g:, 'airline#extensions#branch#notexists', g:airline_symbols.notexists)
+  if !has_key(s:untracked_git, a:file)
+    if s:has_async
+      call s:DoAsync(s:git_cmd, a:file)
+    else
+      let output = system(s:git_cmd. shellescape(a:file))
+      if output[0:1] is# '??' && output[3:-2] is? a:file
+        let untracked = get(g:, 'airline#extensions#branch#notexists', g:airline_symbols.notexists)
+      endif
+      let s:untracked_git[a:file] = untracked
     endif
-    let s:untracked_git[a:file] = untracked
   endif
-  return untracked
 endfunction
 
 function! s:get_hg_untracked(file)
-  if s:has_lawrencium
-    " delete cache when unlet b:airline head?
-    let untracked = ''
-    if empty(a:file)
-      return untracked
-    endif
-    if has_key(s:untracked_hg, a:file)
-      let untracked = s:untracked_hg[a:file]
+  if !s:has_lawrencium || empty(a:file) || !executable('hg')
```

Comment:
You check for lawrencium here, but you don't use it and you don't check for fugitive in get_git_untracked.

Consider either:
* Removing the check for has_lawrencium. It is not used therefore there's no need to check for it here.
* Adding the has_fugitive check to get_git_untracked. It would maintain consistency.